cmd/repository/json-pkg-registry.go

```diff
@@ -0,0 +1,85 @@
+package repository
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/criteo/command-launcher/internal/command"
+)
+
+type jsonRegistry struct {
+	defaultRegistry
+	pathname string
+}
+
+func newJsonRegistry(path string) (Registry, error) {
+	reg := jsonRegistry{
+		defaultRegistry: defaultRegistry{
+			packages:       make(map[string]command.PackageManifest),
+			groupCmds:      make(map[string]command.Command),
+			executableCmds: make(map[string]command.Command),
+		},
+		pathname: path,
+	}
+
+	return &reg, nil
+}
+
+func (reg *jsonRegistry) Load(_ string) error {
+	_, err := os.Stat(reg.pathname)
+	if !os.IsNotExist(err) {
+		payload, err := os.ReadFile(reg.pathname)
+		if err != nil {
+			return err
+		}
+
+		packages := make(map[string]*defaultRegistryEntry, 0)
+		if err = json.Unmarshal(payload, &packages); err != nil {
+			return err
+		}
+
+		for name, pkg := range packages {
+			reg.packages[name] = pkg
+		}
+	}
+
+	reg.extractCmds()
+
+	return nil
+}
+
+func (reg *jsonRegistry) store() error {
```

Comment:
By putting the store method private, we have to make a persistence for every operation. It makes the code less flexible. I would keep it public so that we can do:
- Add
- Add
- Remove
- Store